kepub/converter.go

```diff
@@ -83,6 +88,43 @@ func (c *Converter) Convert(dir string) error {
 	return nil
 }
 
+func (c *Converter) GetHtmlFiles(dir string) ([]string, error) {
+	opfPath, err := FindOPF(dir)
+	if err != nil {
+		return []string{}, fmt.Errorf("transform opf: find: %w", err)
+	}
+
+	rsk, err := os.Open(opfPath)
+	if err != nil {
+		return []string{}, fmt.Errorf("error opening opf file: %w", err)
+	}
+	defer rsk.Close()
+
+	doc := etree.NewDocument()
+	if _, err = doc.ReadFrom(rsk); err != nil {
+		return []string{}, fmt.Errorf("error parsing opf file: %w", err)
+	}
+
+	var files []string
+
+	elements := doc.FindElements("//package/manifest/item")
+	for _, element := range elements {
+		href := ""
+		mimeType := ""
+		for _, attr := range element.Attr {
+			if attr.Key == "media-type" {
+				mimeType = attr.Value
+			} else if attr.Key == "href" {
+				href = attr.Value
+			}
+		}
+		if (mimeType == "text/html" || mimeType == "application/xhtml+xml") && href != "" {
```

Comment:
Is there a reason why you included the `href` check?

Reply:
Instead, can you `os.Stat` the file and skip it if `os.IsNotExist`, and otherwise return an error (which includes the filename) if `err != nil`? This is to preserve the previous more-lenient behaviour as much as possible.

Reply:
> Is there a reason why you included the `href` check?

To prevent errors if the item is malformed and `href` is missing.